### ISSUE TITLE
Improved readability for emotes

### ISSUE THREAD

**Issue by Irarara** [RPTools/maptool#2271]
**Is your feature request related to a problem? Please describe.**
Presently all emotes (/emote and /emotes) show up in green making them rather hard to tell apart when looking through chat and easy to miss.

**Describe the solution you'd like**
At the very least, I think this could be improved by making them use the colors set by players (as with /say and /ooc) instead of a uniform green. My group has played for a long time with a build that does this and have found it to be a great improvement in terms of readability. 

However, with the change to /say output that came with 1.7.0, I think it might be nicer to simply style them after /say, token image and all (when they're enabled). Something like:

![Example emotes](https://user-images.githubusercontent.com/59770924/96748698-e6512880-1397-11eb-905a-425b13ec3d0a.png)

This does come at the cost of emotes taking up a bit more space in chat, but the presence of the image should make them even more identifiable. But, if some find this undesirable perhaps it could be optional?

**Comment by Azhrei** [comment]
Sounds like a fine addition to me.

**Comment by Phergus** [comment]
I've never used the emote function but it seems like a reasonable change to me.

**Comment by Phergus** [comment]
Tested.  The one potential issue I see is that emotes don't stand out as much.  I wonder if emote output shouldn't stay with a different color or perhaps just be in italics.

**Comment by Irarara** [comment]
I can only speak from personal experience, but I haven't found that to be an issue from playing with emotes like this. My group plays over text, and distinguishing /say and /emote as a glance is less important for us than being able to quickly determine the source of the message and having them stand out as a whole. Avatars help with this a lot on their own, but I think the text color helps reinforce it too.

That said, that's all personal to my group. If others find it's making things _more_ difficult to follow I agree something like italics would make sense. Or maybe styling it differently somehow, like changing the border color (between the avatar and message) or something?

**Comment by Phergus** [comment]
Well you've already solved the identifying the source of the message problem with the addition of the avatar so that is no longer a concern.  As a GM I would think it important to be able to tell at a glance that something was an emote versus spoken out loud and thus potentially heard by others.

While on the subject, what about the OOC output?  Should it get the same upgrade to include the avatar?

**Comment by Irarara** [comment]
Well, I still think having both is nice (but yes, the avatar basically does solves it on its own), but that's a good point! In my experience, there's lots of instances of spoken dialogue included in emotes, so usually you wouldn't really know or not until you've read it anyway, but I can definitely imagine some finding the distinction useful for reasons like that.

Just cause this got me thinking and I wanted to see how it'd look, how would you feel about something like this? Too subtle?
![java_2020-12-13_11-35-32](https://user-images.githubusercontent.com/59770924/102018200-c146b980-3d39-11eb-8a6b-f9479eb22d03.png)

Regarding /ooc, I think it's good as is but I don't feel super strongly about it. The text color does a good job quickly communicating the source of the message and there isn't much need to care about who is being impersonated (I think?). That's one where if it included avatars I'd care more about it looking distinct.

**Comment by Phergus** [comment]
I'd say that looks pretty good and is easily distinguishable.

re: OOC - as it is _out-of-character_ I guess it doesn't really make sense to show a character avatar and should probably instead have the player name.  Don't know if anyone ever uses it so it may not matter.  It just sticks out as being different.

**Comment by Irarara** [comment]
Yeah, I agree it is kind of funny /ooc doesn't just use the player name, current identity doesn't really make sense for it. I actually thought about bringing that up, but it seemed like a separate thing. I think it'd be a good thing to change though.

This is going a bit far afield now, but whisper has similar behavior, in that the message you get when receiving a whisper shows the name of the current identity. I suppose if you think of it as an IC thing it makes sense, but whisper can be either and is probably more often OOC (and it's kind of funny since the sender will always see the name of the player they're sending to, so it's different whether you're receiving or sending.)

In any case, I like how that looks too, so if that's good I'll go ahead and open a PR. Should I go ahead with any of the other stuff we've been talking about (player names vs current identitiy)? Or should a separate issue be opened?

**Comment by Phergus** [comment]
Go ahead with the pr for emote change.

For ooc and whisper, a new issue to explore options for those would be best.

**Comment by Phergus** [comment]
I like it!  Just enough to be obvious without being distracting.
